depends: Consistent use of package variable

Bushstar opened the pull request:
All other mk files use the package variable consistently except for the three instances here, which have always been here, since depends was introduced in 0.10.

fanquake commented on depends/packages/boost.mk:
```diff
@@ -29,11 +29,11 @@ $(package)_cxxflags_android=-fPIC
 
 define $(package)_preprocess_cmds
-  echo "using $(boost_toolset_$(host_os)) : : $($(package)_cxx) : <cxxflags>\"$($(package)_cxxflags) $($(package)_cppflags)\" <linkflags>\"$($(package)_ldflags)\" <archiver>\"$(boost_archiver_$(host_os))\" <striper>\"$(host_STRIP)\"  <ranlib>\"$(host_RANLIB)\" <rc>\"$(host_WINDRES)\" : ;" > user-config.jam
+  echo "using $($(package)_toolset_$(host_os)) : : $($(package)_cxx) : <cxxflags>\"$($(package)_cxxflags) $($(package)_cppflags)\" <linkflags>\"$($(package)_ldflags)\" <archiver>\"$(boost_archiver_$(host_os))\" <striper>\"$(host_STRIP)\"  <ranlib>\"$(host_RANLIB)\" <rc>\"$(host_WINDRES)\" : ;" > user-config.jam
```
There's also `$(boost_archiver_$(host_os))` here.

Bushstar replied:
Thanks. Now updated and force pushed.

Bushstar committed:
depends: Consistent use of package variable

All other mk files use the package variable consistently except for the two instances here, which have always been here, since depends was introduced in 0.10.

fanquake reviewed (approved):
ACK 22c5a986e95d2bd14273465ca0e15fbe3772252d - tested a `make boost -C depends/ -j8`.
